swagger/smart_transfer_protocol_ws.yaml

```diff
@@ -0,0 +1,421 @@
+      security:
+        - jwt: []
+      message:
+        messageId: odf.dataset.DatasetPushComplete
+        name: DatasetPushComplete
+        contentType: application/json
+        payload:
+          type: object
+          required:
+            - datasetId
+            - pushRequestId
+          datasetId:
+            type: string
+            format: did
+            description: Dataset identifier       
+          pushRequestId:
+            type: string
+            format: uuid
+            description: Request identifier of the started push flow            
+
+  dataset/push/complete/confirmed:
+    publish:
+      operationId: odf.dataset.push.complete.confirmed
+      summary: Informs consumer the push completion is confirmed
+      security:
+        - jwt: []
+      message:
+        messageId: odf.dataset.DatasetPushCompleteConfirmed
+        name: DatasetPushCompleteConfirmed
+        contentType: application/json
+        payload:
+          type: object
+          required:
+            - datasetId
+            - pushRequestId
+            - newHead
+          datasetId:
+            type: string
+            format: did
+            description: Dataset identifier          
+          pushRequestId:
+            type: string
+            format: uuid
+            description: Request identifier of the started push flow
+          newHead:
+            type: string
+            format: multihash
+            description: New dataset head block hash after completion of this push request
+
+  dataset/error:
+    publish:
+      operationId: odf.dataset.error
+      summary: Informs about error happened during one of API operations with the dataset
+      security:
+        - jwt: []
+      message:
+        messageId: odf.dataset.DatasetError
+        name: DatasetError
+        contentType: application/json
+        payload:
+          type: object
+          required:
+            - datasetId
+            - errorDetails
+          properties:
+            datasetId:
+              type: string
+              format: did
+              description: Dataset identifier
+            errorDetails:
+              $ref: '#/components/schemas/ErrorDetails'
+
+components:
+  schemas:
+    TarOfBlocks:
+      type: object
+      description: An archive of blocks segment being transferred
+      required:
+        - blockCount
+        - blockData
+      properties:
+        blockCount:
+          type: integer
+          minimum: 0
+        blockData:
+          type: string
+          contentEncoding: base64
+          contentMediaType: "application/tar"
+
+    NegotiatedDownload:
+      type: object
+      required:
+        - dataFile
+        - downloadNegotiationStatus
+      properties:
+        dataFile:
+          $ref: '#/components/schemas/DataFileReference'
+        downloadNegotiationStatus:
+          type: string
+          enum: [UnexpectedFile, ProceedWithDownload]
```

Comment:
`UnexpectedFile` seems like an error that should terminate the entire operation, is it not?

Reply:
Removed the status and made `downloadFrom` mandatory